social_engagement/models.py

```diff
@@ -103,14 +109,19 @@ def generate_leaderboard(cls, course_key, count=3, exclude_users=None, org_ids=N
 
         if org_ids:
             queryset = queryset.filter(user__organizations__in=org_ids)
-        queryset = queryset.values(
-            'user__id',
-            'user__username',
-            'user__profile__title',
-            'user__profile__avatar_url',
-            'score')\
-            .order_by('-score', 'modified')[:count]
-        return queryset
+
+        aggregates = queryset.aggregate(Avg('score'))
+        course_avg = aggregates['score__avg']
+        if count:
```

Comment:
Do we really need an if condition here, when the default value of count is set to 3?